packages/cli-kit/src/public/node/node-package-manager.integration.test.ts

```diff
@@ -4,27 +4,31 @@ import {joinPath, dirname} from './path.js'
 import {describe, expect, test} from 'vitest'
 
 describe('installNPMDependenciesRecursively', () => {
-  test('runs install in all the directories containing a package.json', async () => {
-    await inTemporaryDirectory(async (tmpDir) => {
-      // Given
-      const rootPackage = joinPath(tmpDir, 'package.json')
-      const webPackage = joinPath(tmpDir, 'web/package.json')
-      const backendPackage = joinPath(tmpDir, 'web/backend/package.json')
+  test(
+    'runs install in all the directories containing a package.json',
+    async () => {
+      await inTemporaryDirectory(async (tmpDir) => {
+        // Given
+        const rootPackage = joinPath(tmpDir, 'package.json')
+        const webPackage = joinPath(tmpDir, 'web/package.json')
+        const backendPackage = joinPath(tmpDir, 'web/backend/package.json')
 
-      await mkdir(dirname(webPackage))
-      await mkdir(dirname(backendPackage))
+        await mkdir(dirname(webPackage))
+        await mkdir(dirname(backendPackage))
 
-      await writeFile(rootPackage, JSON.stringify({}))
-      await writeFile(webPackage, JSON.stringify({}))
-      await writeFile(backendPackage, JSON.stringify({}))
+        await writeFile(rootPackage, JSON.stringify({}))
+        await writeFile(webPackage, JSON.stringify({}))
+        await writeFile(backendPackage, JSON.stringify({}))
 
-      // When
-      await expect(
-        installNPMDependenciesRecursively({
-          directory: tmpDir,
-          packageManager: 'pnpm',
-        }),
-      ).resolves.toBe(undefined)
-    })
-  })
+        // When
+        await expect(
+          installNPMDependenciesRecursively({
+            directory: tmpDir,
+            packageManager: 'pnpm',
+          }),
+        ).resolves.toBe(undefined)
+      })
+    },
+    {retry: 3},
```

Comment:
I'd consider turning this test into a unit test that ensures "install" is called for each of the sub-directories that contains a `package.json`. In its current state where we are doing actual invocations to the package managers, it's easy that flakiness happens because those package managers have global states (caches, configuration).